File: backend/src/main/java/com/back/domain/adoption/dto/response/ApplicationSimpleListResponseDto.java
package com.back.domain.adoption.dto.response;

import com.back.domain.adoption.entity.Adoption;
import com.back.domain.adoption.enums.RequestStatus;
import com.back.domain.care.entity.Care;
import com.back.domain.pet.dto.response.PetInfoResponseDto;
import java.time.LocalDateTime;
import lombok.Builder;

@Builder
public record ApplicationSimpleListResponseDto(
    Long id,
    String title,
    String type, // "ADOPTION" 또는 "CARE"
    RequestStatus status,
    LocalDateTime createdAt,
    PetInfoResponseDto petInfo, // Pet 정보 추가
    LocalDateTime desiredStartDate, // Care인 경우에만 사용
    LocalDateTime desiredEndDate    // Care인 경우에만 사용
) {
    
    public static ApplicationSimpleListResponseDto fromAdoption(Adoption adoption) {
        return ApplicationSimpleListResponseDto.builder()
                .id(adoption.getId())
                .title(adoption.getTitle())
                .type("ADOPTION")
                .status(adoption.getStatus())
                .createdAt(adoption.getCreatedAt())
                .petInfo(PetInfoResponseDto.from(adoption.getPet()))
                .build();
    }
    
    public static ApplicationSimpleListResponseDto fromCare(Care care) {
        return ApplicationSimpleListResponseDto.builder()
                .id(care.getId())
                .title(care.getTitle())
                .type("CARE")
                .status(care.getStatus())
                .createdAt(care.getCreatedAt())
                .petInfo(PetInfoResponseDto.from(care.getPet()))
                .desiredStartDate(care.getDesiredStartDate())
                .desiredEndDate(care.getDesiredEndDate())
                .build();
    }
} 

File: backend/src/main/java/com/back/domain/adoption/dto/response/ApplicationSimpleListResponseDto.kt
package com.back.domain.adoption.dto.response

import com.back.domain.adoption.entity.Adoption
import com.back.domain.adoption.enums.RequestStatus
import com.back.domain.care.entity.Care
import com.back.domain.pet.dto.response.PetInfoResponseDto
import java.time.LocalDateTime

data class ApplicationSimpleListResponseDto(
    val id: Long?,
    val title: String,
    val type: String,  // "ADOPTION" 또는 "CARE"
    val status: RequestStatus,
    val createdAt: LocalDateTime?,
    val petInfo: PetInfoResponseDto,  // Pet 정보 추가
    val desiredStartDate: LocalDateTime?,  // Care인 경우에만 사용
    val desiredEndDate: LocalDateTime? // Care인 경우에만 사용
) {
    companion object {
        fun fromAdoption(adoption: Adoption): ApplicationSimpleListResponseDto {
            return ApplicationSimpleListResponseDto(
                id = adoption.id,
                title = adoption.title,
                type = "ADOPTION",
                status = adoption.status,
                createdAt = adoption.createdAt,
                petInfo = PetInfoResponseDto.from(adoption.pet),
                desiredStartDate = null,
                desiredEndDate = null
            )
        }

        fun fromCare(care: Care): ApplicationSimpleListResponseDto {
            return ApplicationSimpleListResponseDto(
                id = care.id,
                title = care.title,
                type = "CARE",
                status = care.status,
                createdAt = care.createdAt,
                petInfo = PetInfoResponseDto.from(care.pet),
                desiredStartDate = care.desiredStartDate,
                desiredEndDate = care.desiredEndDate
            )
        }
    }
}